app/uk/gov/hmrc/economiccrimelevyaccount/controllers/ErrorHandler.scala

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.economiccrimelevyaccount.controllers
+
+import cats.data.EitherT
+import play.api.Logging
+import uk.gov.hmrc.economiccrimelevyaccount.models.errors.{BadGateway, DesSubmissionError, IntegrationFrameworkError, InternalServiceError, ResponseError}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+trait ErrorHandler extends Logging {
```

Comment:
On the controllers, are we testing the possible error scenarios and how they are handled by this trait?

I've had a look and in one we are only testing happy paths, and in the others it's a mixture. We should be covering all response types we can return in the controllers.

I understand we have another ticket to handle test improvements, so if you plan to have that covered there. Please resolve this comment.

Reply:
Yes I did realise that and planned for it to be tackled in the ticket for adding tests.